pkg/resource/container.go

```diff
@@ -0,0 +1,302 @@
+package resource
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/leaktk/scanner/pkg/fs"
+	"github.com/leaktk/scanner/pkg/logger"
+	"io"
+	iofs "io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/pkg/blobinfocache"
+	"github.com/containers/image/v5/types"
+)
+
+// Container provides a pull and hold a container
+type Container struct {
+	// Provide common helper functions
+	BaseResource
+	clonePath string
+	location  string
+	options   *ContainerOptions
+}
+
+// ContainerOptions are options for the Container resource
+type ContainerOptions struct {
+	Local      bool     `json:"local"`
+	Exclusions []string `json:"exclusions"`
+	Arch       string   `json:"arch"`
+}
+
+// NewContainer returns a configured Container resource for the scanner to scan
+func NewContainer(location string, options *ContainerOptions) *Container {
+	return &Container{
+		location: location,
+		options:  options,
+	}
+}
+
+// Kind of resource (always returns Container here)
+func (r *Container) Kind() string {
+	return "Container"
+}
+
+// String representation of the resource
+func (r *Container) String() string {
+	return r.location
+}
+
+// Clone the resource to the desired local location and store the path
+func (r *Container) Clone(path string) error {
+	r.clonePath = path
+	if r.options.Local {
+		return r.cloneLocalResource(path, r.location)
+	}
+	return r.cloneRemoteResource(path, r.location)
```

Comment:
Would it make sense to use `copy.Image` ([example](https://github.com/containers/buildah/blob/main/internal/source/pull.go#L74)) if `!r.options.Local` and then treat both scans the same?

Basically:

- If local is false
  - set the storage location to the clone path
  - `copy.Image`
- scan the image[1]   (maybe using something from the github containers repo that doesn't care where the storage is)
- if not local storage, blow away the clone path during the cleanup


[1] I'm thinking that should handle a bunch of different cases and we can lean pretty heavily on the containers code for handeling auth, image formats, and all kinds of stuff.

Thoughts?

Reply:
To close this out. The existing method will be used for remote sources as it is the quickest with least dependency on a valid docker/podman installation. 

For local we will be gating it for Linux only and relying more heavily on a valid docker/podman setup.